scripts/diff-workflow-against-master.sh

```diff
@@ -0,0 +1,471 @@
+#  NETRC is currently the most secure way to provide
+#  credentials to CURL, because it prevents other processes
+#  from inspect the process environment and cmd arguments;
+#  both of which can be found in /proc).
+#
+function init_netrc() {
+	# Check and setup the GitHub personal access token
+	local netrc="$HOME/.netrc"
+	if grep -q "^machine $authority" "$netrc"; then
+		echo "Found credentials for $authority in $netrc"
+		return
+	fi
+	# Get the username
+	clear
+	echo ""
+	echo "1. Enter your GitHub account email address, example: jcousteau@scuba.fr"
+	echo ""
+	read -r -p "GitHub account email: " username
+
+	# Get the token
+	clear
+	echo ""
+	echo "2. Login to GitHub and visit https://github.com/settings/tokens"
+	echo ""
+	echo "3. Click 'Generate new token' and set the 'public_repo' permission:"
+	echo ""
+	echo "    [ ]  repo                  Full control of private repos"
+	echo "        [ ] repo:status        Access commit status"
+	echo "        [ ] repo_deployment    Access deployment status"
+	echo "        [X] public_repo        Access public repositories"
+	echo "        [ ] repo:invite        Access repository invitations"
+	echo ""
+	echo "   Type a name for the token then scroll down and click 'Generate Token'."
+	echo ""
+	echo "4. Copy & paste your token, for example: f7e6b2344bd2c1487597b61d77703527a692a072"
+	echo ""
+	# Deliberately echo the token so the user can verify its correctness
+	read -r -p "Personal Access Token: " token
+
+	# Add the credential to netrc
+	clear
+	if [[ -n "${username:-}" && -n "${token:-}" ]]; then
+		{
+			echo "machine $authority"
+			echo "login $username"
+			echo "password $token"
+			echo ""
+		} >> "$netrc"
+		echo "Added your credentials to $netrc"
+		# Ensure netrc is only readable by the user
+		chmod 600 "$netrc"
+	else
+		echo "Failed to setup $netrc or some of the credentials were empty!"
+		exit 1
+	fi
+}
+
+##
+#  Makes strings suitable for directory and filenames
+#   - spaces => underscores
+#   - upper-case => lower-case
+#   - slashes => dashes
+#   - parenthesis => stripped
+#   - equals  => dashes
+#
+function sanitize_name() {
+	local name="$1"
+	echo "$name" | sed 's/\(.*\)/\L\1/;s/ /_/g;s^/^-^g;s/[()]//g;s/=/-/g'
+}
+
+##
+#  Returns how old a file or directory is, in seconds.
+#
+function seconds_old() {
+	echo $(( "$(date +%s)" - "$(stat -L --format %Y "$1")" ))
+}
+
+##
+#  Creates a storage area for all content fetched by the script.
+#  This include cached JSON output (valid for 5 minutes), along
+#  with zip assets, log files, and diffs.
+#
+declare -g parent # used by the trap
+function init_dirs() {
+	local repo_postfix
+	repo_postfix="$(basename "$repo")"
+	parent="/tmp/$repo_postfix-workflows-$USER"
+	assets_dir="$parent/assets"
+	cache_dir="$parent/cache"
+	diffs_dir="$parent/diffs"
+	logs_dir="$parent/logs"
+	declare -gr assets_dir
+	declare -gr cache_dir
+	declare -gr diffs_dir
+	declare -gr logs_dir
+	echo "Initializing storage area: $parent"
```

Comment:
It took me a while to figure out where the logs were actually downloaded (I missed this line when reading output) - can you remove this echo from here, place it just before program exits (so it'll be the last line of output) and give make the message less technical, e.g.: "Copy of logs in: <path>".

Reply:
Will do; much more convenient!

Reply:
Added.